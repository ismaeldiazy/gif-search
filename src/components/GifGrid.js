import React from 'react'

export const GifGrid = ({ category }) => {
    
    const getGifs = async() => {

        const url = 'https://api.giphy.com/v1/gifs/search?q=cheeseburger&limit=10&api_key=e4Rbc8Hq9DMMjYD6rbhsFUnRdqoxGorD'
        const response = await fetch(url);
        const { data } = await response.json();

        const gifs = data.map( img => {
            return {
                id: img.id,
                title: img.title,
                url: img.images?.downsized_medium.url
            }
        });

        console.log(gifs);
    }

    getGifs();

    return (
        <div>
            <h3> { category } </h3>
        </div>
    )
}
