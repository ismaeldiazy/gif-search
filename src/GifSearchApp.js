import React, { useState } from 'react';
import { AddCategory } from './components/AddCategory';
import { GifGrid } from './components/GifGrid';
import { DiReact } from "react-icons/di"; 

export const GifSearchApp = ( ) => {
    // const categories = ['One punch', 'Samurai X', 'Dragon ball'];
    const [categories, setCategories] = useState( ['Berserk'] );

    // const handleAdd = () => {
    //     // setCategories( ['Attack on Titan', ...categories]);
    //     setCategories(cats => [...cats, 'Attack on Titan']);
    // }

    return (
        <>  
            <div className="header-block">
                <div className="title-block">
                    <h2>GifSearchApp</h2>
                    <DiReact className="reactIcon"/>
                </div>
                <AddCategory setCategories={ setCategories }/>
            </div>


            <ol>
                { 
                    categories.map( category => 
                        <GifGrid 
                            key= { category }
                            category={ category } 
                        />  
                    )
                }
            </ol>
        </>
    );
}

export default GifSearchApp;