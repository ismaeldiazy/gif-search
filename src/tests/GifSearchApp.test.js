import { shallow } from 'enzyme';
import React from 'react';
import { GifGrid } from '../components/GifGrid';
import { GifSearchApp } from '../GifSearchApp';

describe('testing GifSearchApp', () => {

    const wrapper = shallow( <GifSearchApp /> )

    test('should match snapshot', () => {
        
        expect( wrapper ).toMatchSnapshot();
    });

    // test('should show a categories list', () => {
    //     const categories = ['One Punch', 'Dragon Ball'];

    //     const wrapper = shallow( <GifSearchApp defaultCategories = {categories}/>)

    //     expect( wrapper ).toMatchSnapshot();
    //     expect( wrapper.find('GifGrid').length).toBe(categories.length);
    // })
    
    
})
