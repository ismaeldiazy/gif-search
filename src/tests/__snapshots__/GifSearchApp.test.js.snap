// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing GifSearchApp should match snapshot 1`] = `
<Fragment>
  <div
    className="header-block"
  >
    <div
      className="title-block"
    >
      <h2>
        GifSearchApp
      </h2>
      <DiReact
        className="reactIcon"
      />
    </div>
    <AddCategory
      setCategories={[Function]}
    />
  </div>
  <ol>
    <GifGrid
      category="Berserk"
      key="Berserk"
    />
  </ol>
</Fragment>
`;
